ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala

```diff
@@ -103,44 +103,88 @@ final class EmberClientBuilder[F[_]: Async] private (
       pushPromiseSupport = pushPromiseSupport,
     )
 
+  /** Sets a custom `TLSContext`.
+    * By default a `TLSContext` is created from the system default `SSLContext`.
+    */
   def withTLSContext(tlsContext: TLSContext[F]) =
     copy(tlsContextOpt = tlsContext.some)
+
+  /** Unset any `TLSContext` and creates one from the system default `SSLContext`. */
   def withoutTLSContext = copy(tlsContextOpt = None)
 
+  /** Sets the `SocketGroup`, a group of TCP sockets to be used in connections. */
   def withSocketGroup(sg: SocketGroup[F]) = copy(sgOpt = sg.some)
 
+  /** Sets the connection pool's total maximum number of idle connections.
+    * Per `RequestKey` values set with `withMaxPerKey` cannot override this total maximum.
+    */
```

Comment:
If folks are happy with the connection pool specific docs, I'll add something similar to upstream [keypool](https://github.com/typelevel/keypool)